review_pr.py

```
@@ -18,37 +32,55 @@
-    return response.json()[0]["number"] if response.status_code == 200 and response.json() else None
+    try:
+        url = f"{GITHUB_API_BASE_URL}/repos/{REPO_NAME}/pulls?state=open&sort=created&direction=desc"
+        response = requests.get(url, headers=GITHUB_HEADERS)
+        response.raise_for_status()
+        prs = response.json()
+        return prs[0]["number"] if prs else None
+    except requests.RequestException as e:
+        logger.error(f"Error fetching latest PR number: {e}")
+        return None
 
- # Fetch PR branch name
+# It fetches the PR branch name anf then use github api to fetch the modified files
 def get_pr_branch(pr_number):
-    url = f"{GITHUB_API_BASE_URL}/repos/{REPO_NAME}/pulls/{pr_number}"
-    response = requests.get(url, headers=GITHUB_HEADERS)
-    return response.json().get("head", {}).get("ref") if response.status_code == 200 else None
+    try:
+        url = f"{GITHUB_API_BASE_URL}/repos/{REPO_NAME}/pulls/{pr_number}"
+        response = requests.get(url, headers=GITHUB_HEADERS)
+        response.raise_for_status()
+        return response.json().get("head", {}).get("ref")
+    except requests.RequestException as e:
+        logger.error(f"Error fetching PR branch: {e}")
+        return None
 
- # Fetch modified files
-def get_modified_files(pr_number):   
-    url = f"{GITHUB_API_BASE_URL}/repos/{REPO_NAME}/pulls/{pr_number}/files"
-    response = requests.get(url, headers=GITHUB_HEADERS)
-    return response.json() if response.status_code == 200 else []
+# It fetches the modified files in the PR using github api
+def get_modified_files(pr_number):
+    try:
+        url = f"{GITHUB_API_BASE_URL}/repos/{REPO_NAME}/pulls/{pr_number}/files"
+        response = requests.get(url, headers=GITHUB_HEADERS)
+        response.raise_for_status()
+        return response.json()
+    except requests.RequestException as e:
+        logger.error(f"Error fetching modified files: {e}")
+        return []
```

In this catch block you should raise the exception or exit the program after logging the error message, to prevent the script from running any further in case of an error.

**Suggested Code:**
```py
exit(1)
```